### PR TITLE
Allen-Cahn equation documentation

### PR DESCRIPTION
**Pull request by ryankrum**

### CODE REVIEW

**Comment by lululxvi** [comment]
@anranjiao Please check. After you think everything is good, then I will double-check.

**Comment by anranjiao** [comment]
I've checked it and everything is good now.

**Comment by lululxvi** [comment]
Change file name to allen.cahn.rst

File: docs/demos/pinn_forward/Allen_Cahn.rst [hunk 1]
```diff
@@ -0,0 +1,125 @@
+Allen-Cahn Equation
+================
+
+Problem Setup
+--------------
+
+We will solve an Allen-Cahn equation with hard initial and boundary conditions:
```
Comment (lululxvi) [inline docs/demos/pinn_forward/Allen_Cahn.rst:7]: We will solve an Allen-Cahn equation:

File: docs/demos/pinn_forward/Allen_Cahn.rst [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+.. code-block:: python
+
+    net = dde.nn.FNN([2] + [20] * 3 + [1], "tanh", "Glorot normal")
+    
+Next, we consider the initial conditions and boundary constraints, defining the transformation of the output and applying it to the network. In this case, ``x^2\cos(\pi x) + t(1 - x^2)y`` is used. When ``t=0``, the initial condition ``x^2\cos(\pi x)`` is satisfied. When ``x=-1`` or ``x=-1``, the boundary condition ``y(-1, t) = y(1, t) = -1`` is satisfied. This demonstrates that the initial condition and the boundary constraint are both hard conditions.
+
+.. code-block:: python
+
+    def output_transform(x, y):
+      return x[:, 0:1]**2 * tf.cos(np.pi * x[:, 0:1]) + x[:, 1:2] * (1 - x[:, 0:1]**2) * y
```
Comment (lululxvi) [inline docs/demos/pinn_forward/Allen_Cahn.rst:72]: The indent here is not correct.

File: docs/demos/pinn_forward/allen.cahn.rst [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    
+The parameter ``num_domain=8000`` is the number of training residual points sampled inside the domain, and the parameter ``num_boundary=400`` is the number of training points sampled on the boundary. We also include the parameter ``num_initial=800``, which represents the number of initial residual points for the initial conditions.
+
+Next, we choose the network. Here, we use a fully connected neural network of depth 4 (i.e., 3 hidden layers) and width 20:
+    
+.. code-block:: python
+
+    net = dde.nn.FNN([2] + [20] * 3 + [1], "tanh", "Glorot normal")
+    
+Next, we consider the initial conditions and boundary constraints, defining the transformation of the output and applying it to the network. In this case, ``x^2\cos(\pi x) + t(1 - x^2)y`` is used. When ``t=0``, the initial condition ``x^2\cos(\pi x)`` is satisfied. When ``x=-1`` or ``x=-1``, the boundary condition ``y(-1, t) = y(1, t) = -1`` is satisfied. This demonstrates that the initial condition and the boundary constraint are both hard conditions.
```
Comment (lululxvi) [inline docs/demos/pinn_forward/allen.cahn.rst:69]: ``x^2\cos(\pi x) + t(1 - x^2)y`` ``x^2\cos(\pi x)`` --- format not corrct. 

See https://deepxde--752.org.readthedocs.build/en/752/demos/pinn_forward/allen.cahn.html

**Pull request merged**